src/cdk/collections/selection-model.ts

```diff
@@ -238,6 +233,21 @@ export class SelectionModel<T> {
   private _hasQueuedChanges() {
     return !!(this._deselectedToEmit.length || this._selectedToEmit.length);
   }
+
+  /** Returns a value that is comparable to inputValue by applying compareWith function, returns the same inputValue otherwise. */
+  private _getConcreteValue(inputValue: T): T {
+    let concreteValue = inputValue;
+    if (!this.compareWith) {
+      return concreteValue;
+    } else {
+      this._selection.forEach(selectedValue => {
+        if (this.compareWith?.call(undefined, inputValue, selectedValue)) {
+          concreteValue = selectedValue;
```

Comment:
Don't we want to break here once we find the first match?

Reply:
Yep !